### DISCUSSION TITLE
Support for setting feature state

### DISCUSSION THREAD

**Discussion by king-d-dev** [rnmapbox/maps#2761] [Ideas]
the ability to set feature state has now been added to the upstream native libraries and i would like to make a bumble request that this feature be added to the react-native version as well. Kindly find the link to confirm the feature state addition to the native libraries here https://github.com/mapbox/mapbox-gl-native/issues/12413#issuecomment-809590260

**Comment by mfazekas** [comment]
See also https://github.com/rnmapbox/maps/issues/997#issuecomment-1078985963

**Comment by saaivs** [comment]
Any plans or estimates to implement [setFeatureState](https://docs.mapbox.com/android/maps/api/10.0.0/mapbox-maps-android/com.mapbox.maps/-map-interface(android-jvm)/set-feature-state.html) support ?

**Comment by philiplindberg** [comment]
I would also love to see this implemented. Apparently there's a draft PR in the works since July of last year (https://github.com/rnmapbox/maps/pull/2942), is that close to being merged? @Gnative are you still looking for help on the Android side?

**Comment by davor-bauk-sh** [comment]
I opened a draft PR to add support for feature state (#3741). I've tested it in my own app where it seems to work well, but I'm having trouble getting it to work in the example app. Any help with this would be much appreciated.

**Comment by davor-bauk-sh** [comment]
My PR is ready for review: #3741
